conan/tools/microsoft/subsystems.py

```diff
@@ -1,174 +1,8 @@
-    cwd_inside = subsystem_path(subsystem, cwd)
-    wrapped_user_cmd = command
-    if env_shell:
-        # Wrapping the inside_command enable to prioritize our environment, otherwise /usr/bin go
-        # first and there could be commands that we want to skip
-        wrapped_user_cmd = environment_wrap_command(env_shell, command,
-                                                    cwd=conanfile.generators_folder)
-    inside_command = 'cd "{cwd_inside}" && ' \
-                     '{wrapped_user_cmd}'.format(cwd_inside=cwd_inside,
-                                                 wrapped_user_cmd=wrapped_user_cmd)
-
-    inside_command = escape_windows_cmd(inside_command)
-
-    final_command = 'cd "{cwd}" && {wrapped_shell} --login -c {inside_command}'.format(
-        cwd=cwd,
-        wrapped_shell=wrapped_shell,
-        inside_command=inside_command)
-    conanfile.output.info('Running in windows bash: %s' % final_command)
-    return conanfile._conan_runner(final_command, output=conanfile.output, subprocess=True)
-
-
-def escape_windows_cmd(command):
-    """ To use in a regular windows cmd.exe
-        1. Adds escapes so the argument can be unpacked by CommandLineToArgvW()
-        2. Adds escapes for cmd.exe so the argument survives cmd.exe's substitutions.
-
-        Useful to escape commands to be executed in a windows bash (msys2, cygwin etc)
-    """
-    quoted_arg = subprocess.list2cmdline([command])
-    return "".join(["^%s" % arg if arg in r'()%!^"<>&|' else arg for arg in quoted_arg])
-
-
-def deduce_subsystem(conanfile, scope):
-    if scope.startswith("build"):
-        if hasattr(conanfile, "settings_build"):
-            the_os = conanfile.settings_build.get_safe("os")
-            subsystem = conanfile.settings_build.get_safe("os.subsystem")
-        else:
-            the_os = platform.system()  # FIXME: Temporary fallback until 2.0
-            subsystem = None
-    else:
-        the_os = conanfile.settings.get_safe("os")
-        subsystem = conanfile.settings.get_safe("os.subsystem")
-
-    if not str(the_os).startswith("Windows"):
-        return None
-
-    if subsystem is None and not scope.startswith("build"):  # "run" scope do not follow win_bash
-        return WINDOWS
-
-    if subsystem is None:  # Not defined by settings, so native windows
-        if not conanfile.win_bash:
-            return WINDOWS
-
-        subsystem = conanfile.conf.get("tools.microsoft.bash:subsystem")
-        if not subsystem:
-            raise ConanException("The config 'tools.microsoft.bash:subsystem' is "
-                                 "needed to run commands in a Windows subsystem")
-    return subsystem
-
-
-def subsystem_path(subsystem, path):
-    """"Used to translate windows paths to MSYS unix paths like
-    c/users/path/to/file. Not working in a regular console or MinGW!
-    """
-    if subsystem is None or subsystem == WINDOWS:
+def unix_path(conanfile, path, subsystem=None):
```

Comment:
I don't see the case for hardcoding the subsystem in recipes, that probably shouldn't be added atm.

Reply:
Wow, I don't know what happened with the rest of the code 😅 .
It is fixed and the test completed.